weather.ha/lib/ha_weather.py

```diff
@@ -0,0 +1,327 @@
+ha_weather_hourly_url_data      = __addon__.getSettingString('ha_weather_hourly_url_data')
+ha_loc_title                    = __addon__.getSettingString('loc_title')
+ha_use_loc_title                = __addon__.getSetting('useHALocName')
+
+# ... and fix them if necessary
+if ha_server and ha_server.endswith('/'):
+    ha_server = ha_server[:-1]
+if ha_weather_url and (not ha_weather_url.startswith('/')):
+    ha_weather_url = '/' + ha_weather_url
+if ha_weather_daily_url and (not ha_weather_daily_url.startswith('/')):
+    ha_weather_daily_url = '/' + ha_weather_daily_url
+if ha_weather_hourly_url and (not ha_weather_hourly_url.startswith('/')):
+    ha_weather_hourly_url = '/' + ha_weather_hourly_url
+
+headers = {'Authorization': 'Bearer ' + ha_key, 'Content-Type': 'application/json'}
+
+WND = None
+
+class MAIN():
+    def __init__(self, *args, **kwargs):
+        log('Home Assistant Weather started.')
+        #self.MONITOR = MyMonitor()
+        mode = kwargs['mode']
+        global WND
+        WND = kwargs['w']
+        self.settings_provided = ha_key and ha_server and ha_weather_url and ha_weather_daily_url and ha_weather_hourly_url
+        if not self.settings_provided:
+            show_dialog(__addon__.getLocalizedString(30010))
+            log('Settings for Home Assistant Weather not yet provided. Plugin will not work.')
+            #raise Exception((ha_key, ha_server, ha_weather_url, ha_weather_daily_url, ha_weather_hourly_url))
+        else:
+            #Test connection / get version
+            response = self.getRequest('/config')
+            if response is not None:
+                log('Response from server: ' + str(response.content))
+                self.getForecasts()
+                #parsedResponse = json.loads(response.text)
+            else:
+                log('Response from server is NONE!')
+                show_dialog('Response from server is NONE!')
+                self.clearProps()
+        log('Home Assistant Weather init finished.')
+
+    def getRequest(self, api_ext):
+        global headers, ha_server
+        retry = 0
+        r = None
+        while retry < MAX_REQUEST_RETRIES:
+            try:
+                log('Trying to make a get request to ' + ha_server + api_ext)
+                r = requests.get(ha_server + api_ext, headers=headers)
+                log('GetRequest status code is: ' + str(r.status_code))
+                if r.status_code == 401:
+                    show_dialog(__addon__.getLocalizedString(30011)) #Error 401: Check your token
+                elif r.status_code == 405:
+                    show_dialog(__addon__.getLocalizedString(30012)) #Error 405: Method not allowed
+                elif r.status_code == 200:
+                    return r
+            except:
+                log('Status code error is: ' + str(r.raise_for_status() if r else 'unknown' ))
+            retry += 1
+            sleep(RETRY_DELAY_S)
```

Comment:
Please don't use `sleep` as it prevents Kodi from shutting down the script. Use the `MyMonitor.waitForAbort(10)` like the other weather add-ons do (https://github.com/xbmc/repo-scripts/blob/26fc0ca904c51d0078b8193c571356dd30b417a8/weather.multi/lib/weather.py#L104C37-L104C54)